### DISCUSSION TITLE
using get_order_buyer_info endpoint

### DISCUSSION THREAD

**Discussion by pkillionFlora** [saleweaver/python-amazon-sp-api#326] [Q&A]
I am utilizing the get_order_buyer_info endpoint and it retrieves the hashed email some of the time and when it does not, it returns "Access to requested resource is denied."

Any ideas on why I experience this only some of the time? Or better yet how to get the buyer_info field when calling getOrders to be populated with the hashed email?

**Comment by saleweaver** [comment]
Amazon has deprecated this endpoint some months ago. You can use get orders like explained in the docs (Orders)

# PII Data

     Orders(restricted_data_token='<token>').get_orders(CreatedAfter=(datetime.utcnow() - timedelta(days=7)).isoformat())

# or use the shortcut

    orders = Orders().get_orders(
    RestrictedResources=['buyerInfo', 'shippingAddress'],
    LastUpdatedAfter=(datetime.utcnow() - timedelta(days=1)).isoformat()
    )

**Reply by pkillionFlora** [reply]
Thank you for the info. So without a restricted data token I cannot reliably retrieve the hashed email information?
